web/src/components/tasks/business-formation/business/BusinessStep.tsx

```diff
@@ -1,69 +1,111 @@
 import { useConfig } from "@/lib/data-hooks/useConfig";
-import { ReactElement, useContext } from "react";
+import { Fragment, ReactElement, useContext, useMemo } from "react";
 
 export const BusinessStep = (): ReactElement => {
   const { Config } = useConfig();
   const { state } = useContext(BusinessFormationContext);
 
+  const isForeign = useMemo(
+    () => state.formationFormData.businessLocationType !== "NJ",
+    [state.formationFormData.businessLocationType]
+  );
+
+  const isNonprofit = state.formationFormData.legalType === "nonprofit";
+
+  const addressSection = (): ReactElement =>
+    isForeign ? <MainBusinessForeignAddressFlow /> : <MainBusinessAddressNj />;
+
+  const lpSection = (): ReactElement | null => {
+    if (state.formationFormData.legalType !== "limited-partnership") return null;
+    return (
+      <>
+        <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
+        <BusinessFormationTextBox
+          maxChars={400}
+          fieldName={"combinedInvestment"}
+          required={true}
+          title={Config.formation.fields.combinedInvestment.label}
+          contentMd={Config.formation.fields.combinedInvestment.body}
+        />
+        <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
+        <BusinessFormationTextBox
+          maxChars={400}
+          fieldName={"withdrawals"}
+          required={true}
+          title={Config.formation.fields.withdrawals.label}
+          contentMd={Config.formation.fields.withdrawals.body}
+        />
+        <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
+        <PartnershipRights />
+        <hr className="margin-bottom-2 margin-top-2" aria-hidden={true} />
+        <BusinessFormationTextBox
+          maxChars={400}
+          fieldName={"dissolution"}
+          required={true}
+          title={Config.formation.fields.dissolution.label}
+          contentMd={Config.formation.fields.dissolution.body}
+        />
+      </>
+    );
+  };
+
+  const businessPurposeSection = (): ReactElement => (
+    <BusinessFormationTextBox
+      maxChars={300}
+      fieldName={"businessPurpose"}
+      required={false}
+      inputLabel={Config.formation.fields.businessPurpose.label}
+      addButtonText={Config.formation.fields.businessPurpose.addButtonText}
+      title={Config.formation.fields.businessPurpose.label}
+      contentMd={Config.formation.fields.businessPurpose.body}
+    />
+  );
+
+  const additionalProvisionsSection = (): ReactElement | null => {
+    if (
+      state.formationFormData.businessLocationType !== "NJ" &&
+      state.formationFormData.legalType !== "foreign-limited-partnership"
+    )
+      return null;
+
+    return <Provisions />;
+  };
+
+  const defaultOrder: (ReactElement | null)[] = [
+    <MainBusiness key={"main-business"} />,
+    <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} key={"business-line-1"} />,
```

Comment:
[pebble] I don't think these `<hr>` elements need a key unless they're in a map.

Reply:
This is kindof nit-picky, but if you wanted, you could move all of the `<hr>`s into the corresponding function call, the same way you have it with `lpSection()`. It might make this a little cleaner.

Reply:
I think I disagree - I think it would be harder to manage because the way the order is changing and things are appearing or disappearing, I feel like it's more likely that you might wind up with doubled-up <hr>s. Idk

Reply:
I think they're necessary because we call `.map` on `elementsToDisplay` which is set to either `defaultOrder` or `nonprofitOrder`.

Reply:
@AnJuHyppolite right, but the key isn't relevant until you're inside the map. You shouldn't need to supply that ahead of time.